MacBry-Maciej-Bryja/SimpleTodoApp1/src/main/java/com/mac/bry/simpleTodo/DAO/TaskDAO.java

```diff
@@ -0,0 +1,133 @@
+				.addAnnotatedClass(User.class)
+				.buildSessionFactory();
+	}
+	
+	@Override
+	public void addTask(Task task) {
+		Session session = factory.getCurrentSession();
+		session.beginTransaction();
+		session.save(task);
+		session.getTransaction().commit();
+		session.close();
+	}
+
+	@Override
+	public void editTaskName(Task task, String newTaskName) {
+		String oldTaskName = task.getTaskName();
+		Session session = factory.getCurrentSession();
+		task.setTaskName(newTaskName);
+		session.getTransaction().commit();
+		System.out.println("You have changed task name from " + oldTaskName + " to " + task.getTaskName());
+	}
+
+	@Override
+	public Task selectTaskByID(int taskID) {
+		Session session = factory.getCurrentSession();
+		session.beginTransaction();
+		Task tempTask = session.get(Task.class, taskID);
+		return tempTask;
+	}
+
+	@Override
+	public void deleteTaskByID(int taskID) {
+		Task taskToDelete = selectTaskByID(taskID);
+		Session session = factory.getCurrentSession();
+		session.delete(taskToDelete);
+		session.getTransaction().commit();
+		session.close();
+	}
+
+	@Override
+	public void changeTaskStatus(Task task) {
+		boolean oldTaskStatus = task.isTaskStatus(); 
+		Session session = factory.getCurrentSession();
+		task.setTaskStatus(!oldTaskStatus);
+		session.getTransaction().commit();
+		System.out.println("You have changed task status from " + oldTaskStatus + " to " + task.isTaskStatus());
+	}
+
+	@Override
+	public List<Task> selectTasksByStatus(boolean status) {
+		int option;
+		if(status==true)
+			option = 1;
+		else option = 0;
+		List<Task> taskList;
+		Session session = factory.getCurrentSession();
+		session.beginTransaction();
+		taskList = session.createQuery("from Task task where task.taskStatus='" + option + "'").getResultList();
+		session.getTransaction().commit();
+		session.close();
+		return taskList;
+	}
+
+	@Override
+	public List<Task> selectTasksByDataOfCompletion(LocalDate dateOfCompletion) {
+		List<Task> taskList;
+		Session session = factory.getCurrentSession();
+		session.beginTransaction();
+		taskList = session.createQuery("from Task task where task.dateOfCompletion='" + dateOfCompletion + "'").getResultList();
+		session.getTransaction().commit();
+		System.out.println("Test1");
+		session.close();
+		return taskList;
+	}
+
+	@Override
+	public List<Task> sortTasksByName() {
+		List <Task> tasksListToSort = getAllTasks();
+		tasksListToSort.sort(new TaskComparatorByName());
+		return tasksListToSort;
+	}
+
+	@Override
+	public List<Task> sortTasksByStatus() {
+		List <Task> tasksListToSort = getAllTasks();
+		tasksListToSort.sort(new TaskComparatorByStatus());;
+		return tasksListToSort;
+	}
+
+	@Override
+	public List<Task> sortTasksByDateOfCompletion() {
+		List <Task> tasksListToSort = getAllTasks();
+		tasksListToSort.sort(new TaskComparatorByDateOfCompletion());
+		return null;
+	}
+
+	@Override
+	public List<Task> getAllTasks() {
+		List <Task> allTasks;
```

Comment:
Definiuj zmiennej jak najpóźniej jest to możliwe.

Reply:
Poprawione